src/Shared/TaskParameter.cs

```diff
@@ -751,6 +756,20 @@ IDictionary ITaskItem2.CloneCustomMetadataEscaped()
                 IDictionary clonedDictionary = new Dictionary<string, string>(_customEscapedMetadata);
                 return clonedDictionary;
             }
+
+            public IEnumerable<KeyValuePair<string, string>> EnumerateMetadata()
+            {
+                if (_customEscapedMetadata == null)
+                {
+                    yield break;
+                }
+
+                foreach (var kvp in _customEscapedMetadata)
+                {
+                    var unescaped = new KeyValuePair<string, string>(kvp.Key, EscapingUtilities.UnescapeAll(kvp.Value));
```

Comment:
It looks like it takes the escaped form if it were available had you not added this. Why does this switch to unescaped?

Reply:
We unify on Unescaped everywhere for logging

Reply:
I wish I didn't have to do this as it made things slower than just returning `_customEscapedMetadata` so I had to write this loop explicitly in two places. Good to be consistently returning unescaped strings everywhere.